doc/sphinx/addendum/type-classes.rst

```diff
@@ -334,6 +340,24 @@ Summary of the commands
 
          This command has no effect when used on a typeclass.
 
+.. _warn-future-coercion-class-field:
+
+   .. warn:: A coercion will be introduced in future versions when using ':>' in 'Class' declarations. Use '#[global] Existing Instance field.' instead if you don't want the coercion.
+
+      .. deprecated:: 8.16
```

Comment:
It's not the warning that is deprecated. You should remove either the `.. warn::` or the `.. deprecated::` in the documentation (keep only one of them). Since it's a bit weird anyway to use the "deprecated" category for something that is going to be added, I'd say just keeping the `.. warn::` seems the best option.